src/Control/Foldl.hs

```diff
@@ -47,6 +47,7 @@ module Control.Foldl (
 
     -- * Folding
     , fold
+    , foldOfWith
```

Comment:
Perhaps you could shorten this name to `foldOf`?  That would be consistent with lens, where the `*Of` family of functions are equivalent to the non-`*Of` versions when you supply `folded` or `traverse` as the lens:

```haskell
fold = foldOf folded
```